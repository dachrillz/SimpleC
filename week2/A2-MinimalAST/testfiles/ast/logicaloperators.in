int simple(){
    a = 5<2;
    b = (1<2);
    return;
}

int order(){
    a = 1 + 5 < 4;
    a = 5 < 4 + 1;
    return;
}

int alloperators(){
    a = b < c;
    b = 5 > d;
    c = a != 5;
    d = g == 2;
    e = 2 >= 2;
    f = 5 <= a;
    return;
}

int several(){
    a = a < a <= a != a == a > a >= a;
    return;
}
