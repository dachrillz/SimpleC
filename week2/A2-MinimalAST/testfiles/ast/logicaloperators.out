Program
  List
    Function ID="simple"
      null
      List
        IdUse ID="a"
          LT
            Var ID="5"
            Var ID="2"
        IdUse ID="b"
          LT
            Var ID="1"
            Var ID="2"
        Return
          null
    Function ID="order"
      null
      List
        IdUse ID="a"
          LT
            Plus
              Var ID="1"
              Var ID="5"
            Var ID="4"
        IdUse ID="a"
          LT
            Var ID="5"
            Plus
              Var ID="4"
              Var ID="1"
        Return
          null
    Function ID="alloperators"
      null
      List
        IdUse ID="a"
          LT
            Var ID="b"
            Var ID="c"
        IdUse ID="b"
          GT
            Var ID="5"
            Var ID="d"
        IdUse ID="c"
          NEQ
            Var ID="a"
            Var ID="5"
        IdUse ID="d"
          EQ
            Var ID="g"
            Var ID="2"
        IdUse ID="e"
          GTE
            Var ID="2"
            Var ID="2"
        IdUse ID="f"
          LTE
            Var ID="5"
            Var ID="a"
        Return
          null
    Function ID="several"
      null
      List
        IdUse ID="a"
          GTE
            GT
              EQ
                NEQ
                  LTE
                    LT
                      Var ID="a"
                      Var ID="a"
                    Var ID="a"
                  Var ID="a"
                Var ID="a"
              Var ID="a"
            Var ID="a"
        Return
          null
