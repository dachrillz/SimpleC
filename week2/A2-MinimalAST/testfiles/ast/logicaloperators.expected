Program
  List
    Function ID="simple"
      null
      List
        IdUse ID="a"
          LT
            Numeral NUMERAL="5"
            Numeral NUMERAL="2"
        IdUse ID="b"
          LT
            Numeral NUMERAL="1"
            Numeral NUMERAL="2"
        Return
    Function ID="order"
      null
      List
        IdUse ID="a"
          LT
            Plus
              Numeral NUMERAL="1"
              Numeral NUMERAL="5"
            Numeral NUMERAL="4"
        IdUse ID="a"
          LT
            Numeral NUMERAL="5"
            Plus
              Numeral NUMERAL="4"
              Numeral NUMERAL="1"
        Return
    Function ID="alloperators"
      null
      List
        IdUse ID="a"
          LT
            Var ID="b"
            Var ID="c"
        IdUse ID="b"
          GT
            Numeral NUMERAL="5"
            Var ID="d"
        IdUse ID="c"
          NEQ
            Var ID="a"
            Numeral NUMERAL="5"
        IdUse ID="d"
          EQ
            Var ID="g"
            Numeral NUMERAL="2"
        IdUse ID="e"
          GTE
            Numeral NUMERAL="2"
            Numeral NUMERAL="2"
        IdUse ID="f"
          LTE
            Numeral NUMERAL="5"
            Var ID="a"
        Return
    Function ID="several"
      null
      List
        IdUse ID="a"
          GTE
            GT
              EQ
                NEQ
                  LTE
                    LT
                      Var ID="a"
                      Var ID="a"
                    Var ID="a"
                  Var ID="a"
                Var ID="a"
              Var ID="a"
            Var ID="a"
        Return
